diff --git a/src/com/transage/privatespace/database/DatabaseAdapter.java b/src/com/transage/privatespace/database/DatabaseAdapter.java
index 07a1be3..8e6ce16 100755
--- a/src/com/transage/privatespace/database/DatabaseAdapter.java
+++ b/src/com/transage/privatespace/database/DatabaseAdapter.java
@@ -9,7 +9,6 @@ import android.database.Cursor;
 import com.transage.privatespace.bean.AppInfo;
 import com.transage.privatespace.bean.CallRecord;
 import com.transage.privatespace.bean.ContactInfo;
-import com.transage.privatespace.bean.People;
 import com.transage.privatespace.bean.SmsInfo;
 import com.transage.privatespace.gallery.photozoom.ImageItem;
 import com.transage.privatespace.utils.ImportExportUtils;
diff --git a/src/com/transage/privatespace/fragment/FG_PrivateSms.java b/src/com/transage/privatespace/fragment/FG_PrivateSms.java
index 75464c1..94b71db 100755
--- a/src/com/transage/privatespace/fragment/FG_PrivateSms.java
+++ b/src/com/transage/privatespace/fragment/FG_PrivateSms.java
@@ -18,13 +18,12 @@ import android.widget.ListView;
 import com.transage.privatespace.R;
 import com.transage.privatespace.activity.PrivateSms;
 import com.transage.privatespace.adapter.SmsListViewAdapter;
-import com.transage.privatespace.bean.People;
+import com.transage.privatespace.bean.ContactInfo;
 import com.transage.privatespace.bean.SmsInfo;
 import com.transage.privatespace.database.DatabaseAdapter;
 import com.transage.privatespace.loader.PrivateSpaceLoader;
 
 import java.util.ArrayList;
-import java.util.List;
 
 /**
  * Created by dongrp on 2016/9/1.
@@ -89,11 +88,12 @@ public class FG_PrivateSms extends BaseFragment {
      */
     public void initData() {
         listSmslist.clear();
-        ArrayList<People> listPrivatePeople = new ArrayList<>();
+        ArrayList<ContactInfo> listPrivatePeople = new ArrayList<>();
         listPrivatePeople.addAll(new DatabaseAdapter(getActivity()).getContacts());
         for(int i = 0;i<listPrivatePeople.size();i++){
 //            Log.e("wangmeng","-----SmsNumberlist-----"+listPrivatePeople.get(i).toString());
-            listSmslist.addAll(new DatabaseAdapter(getActivity()).getLastSms(listPrivatePeople.get(i).getPhoneNum()));
+            listSmslist.addAll(new DatabaseAdapter(getActivity()).
+                    getLastSms(formatNumber(listPrivatePeople.get(i).getPhones().get(0).phoneNumber)));
         }
 //
     }
@@ -108,6 +108,16 @@ public class FG_PrivateSms extends BaseFragment {
         initData();
         initViewAndAdapter();
     }
+    //格式化电话号码 add by wangmeng 20170821
+    private String formatNumber(String number) {
+        String phoneNumStr = null;
+        if (number.contains("+86")) {
+            phoneNumStr = number.replace("+86", "").replaceAll(" ", "").replaceAll("-", "").trim();
+        } else {
+            phoneNumStr = number.replaceAll(" ", "").replaceAll("-", "").trim();
+        }
+        return phoneNumStr;
+    }
 //    public void initData() {
 //        // 1.先获取到所有私密联系人
 //        ArrayList<People> listPrivatePeople = new ArrayList<People>();
